ci/templates/ci/reports.html

```diff
@@ -100,42 +120,64 @@ <h3 class="reports__heading">Report Tasks</h3>
 
     $(document).ready(function() {
-
-      if ($("#id_email_to").val() != '') {
-        var emails = $("#id_email_to").val().split(",");
-        $("#id_email_to").val('');
-        $.each(emails,function(i){
-          $("#id_email_to").after($(".email-to .email-to-group").clone().addClass("field-clone"));
-          $(".field-clone .email-fields").each(function(i){
-            $(this).attr('value', emails[i]);
+      // Replace encoded quotes with double quotes to make valid json
+      var forms = JSON.parse("{{ forms.keys }}".replace(/&#39;/g, '\"'))
+
+      $.each(forms, function(k){
+        // Get form and email_to field ids
+        var form_id = "#".concat(forms[k]);
+        var prefix = forms[k].replace("_form", "");
+        var email_to_id = "#".concat(prefix, "_email_to");
+
+        if ($(email_to_id).val() != '') {
+          var emails = $(email_to_id).val().split(",");
+          $(email_to_id).val('');
+          $.each(emails,function(i){
+            $(email_to_id).after($(".email-to .email-to-group").clone().addClass("field-clone"));
+            $(form_id).find(".field-clone .email-fields").each(function(i){
+              $(this).attr('value', emails[i]);
+            });
           });
-        });
         }
-      else {
-        $("#id_email_to").after($(".email-to .email-to-group").clone().addClass("field-clone"));
-      };
+        else {
+          $(email_to_id).after($(".email-to .email-to-group").clone().addClass("field-clone"));
+        };
 
-      $("#id_email_to").addClass("hide")
-      $(".field-clone:first .remove").replaceWith($(".email-to .add-more").show());
+        $(email_to_id).addClass("hide")
+        $(form_id).find(".field-clone:first .remove")
+            .replaceWith($(".email-to .add-more").clone().show());
+      });
 
       $(".add-more").click(function(){
-          $(".field-clone:last").after($(".email-to .email-to-group").clone());
+          $(this).parents(".control-group").after($(".email-to .email-to-group").clone());
       });
 
       $("body").on("click",".remove",function(){
           $(this).parents(".control-group").remove();
       });
 
       $("body").on("click",".generate",function(){
-        $(".form-horizontal .email-fields").each(function(){
+        var id_prefix = $(this).parents(".form-horizontal").attr("id").replace("_form", "");
+        $(this).parents(".form-horizontal").find(".email-fields").each(function(){
           if ($(this).val() == '')
             $(this).parents(".form-horizontal .control-group").remove();
-            $("#id_email_to").val($(".form-horizontal .email-fields").map( function() {
+            $("#".concat(id_prefix, "_email_to")).val($(".form-horizontal .email-fields").map( function() {
                 return this.value;
               }).get().join(","));
         });
       });
 
+      $("#report_type").on("change", function(){
+        var dropdown = $("#report_type");
+        var form = $("#"+dropdown.val());
```

Comment:
Curious  on what is value of var form? seems like it might be 2 #(hashes)?

Reply:
At this point the value of var form is the form element that has the same id as the value of the selected option in the dropdown.
(as I understand jquery selectors `#element_id` gets the element that has an id of `element_id`, right?)

Reply:
👍